runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsService.java

```diff
@@ -86,6 +93,10 @@ else if (config.highlightNonClanMembers() && !isClanMember)
 			{
 				consumer.accept(player, config.getNonClanMemberColor());
 			}
+			else if ( config.randomizeColor() )
```

Comment:
```suggestion
			else if (config.randomizeColor())
```

Reply:
Also, this behavior seems very weird. If a player were to have only friends selected to be highlighted, enabling this config would cause all non-friend players to be highlighted using a random color, which seems wrong.